### PR TITLE
Add TestGrid config

### PR DESCRIPTION
**Pull request by tehcyx**
**Description**
For jobs specified in the k8s kyma testgrid config (https://github.com/kubernetes/test-infra/pull/14887), to try out initial integration with k8s TestGrid

Changes proposed in this pull request:
- add all cleaners to testgrid
- add release 1.4, 1.5 and 1.6 to testgrid
- add weekly and nightly cluster to testgrid
- add one presubmit (gke-upgrade)
- add one postsubmit (gke-upgrade)
- add one incubator project (hydroform)

**Related issue(s)**
#832

### CODE REVIEW

**Comment by tehcyx** [comment]
PR on k/test-infra got merged.

**Comment by michal-hudy** [comment]
Can you also add https://github.com/kyma-project/test-infra/blob/master/prow/jobs/test-infra/test-infra-kind.yaml#L78 as this is the only job that produces JUnit report in Kyma?

File: prow/config.yaml [hunk 1]
```diff
@@ -526,6 +526,8 @@ deck:
   spyglass:
     size_limit: 500000000 # 500MB
     gcs_browser_prefix: https://gcsweb.build.kyma-project.io/gcs/
+    testgrid_config: gs://k8s-testgrid/config
```
Comment (michal-hudy) [inline prow/config.yaml]: As far as I know, after recent changes, it must be added to https://github.com/kyma-project/test-infra/blob/master/templates/templates/prow-config.yaml

...and yes, I think it makes everything more complicated...

Comment (tehcyx) [reply prow/config.yaml]: Good find, I'll change that.

**Pull request merged**